safe-global/safe-wallet-web
# fix: Move sign msg modal to new flow

## What it solves

Part of #2067 

## How this PR fixes it

- Ports the SignMsgModal (Off-chain signing) to the new tx flow
- Closes the modal when creating a safe apps tx

## How to test it

1. Open a Safe
2. Create a sign message request (e.g. walletconnect safe app using the demo eip1271 app)
3. Observe the new sign message flow
4. Press cancel
5. Observe seeing the confirmation tooltip
6. Confirm the cancellation
7. Observe the tx modal closing
8. Navigate to the Messages tab
9. Sign an existing message
10. Observe the same tx flow
11. Press cancel
12. Observe the tx modal closing
13. Press sign
14. Observe being able to sign the message

## Screenshots
<img width="1225" alt="Screenshot 2023-06-14 at 15 47 29" src="https://github.com/safe-global/safe-wallet-web/assets/5880855/06295c64-dd29-46b4-92b7-5087ebd9a08e">

## Checklist
